qiskit_optimization/algorithms/qrao/quantum_random_access_optimizer.py

```diff
@@ -0,0 +1,304 @@
+    def max_vars_per_qubit(self, max_vars_per_qubit: int) -> None:
+        """Set the maximum number of variables per qubit."""
+        self._max_vars_per_qubit = max_vars_per_qubit
+
+    @property
+    def rounding_scheme(self) -> RoundingScheme:
+        """Return the rounding scheme."""
+        return self._rounding_scheme
+
+    @rounding_scheme.setter
+    def rounding_scheme(self, rounding_scheme: RoundingScheme) -> None:
+        """Set the rounding scheme."""
+        if rounding_scheme is None:
+            rounding_scheme = SemideterministicRounding()
+        self._rounding_scheme = rounding_scheme
+
+    def get_compatibility_msg(self, problem: QuadraticProgram) -> str:
+        """Checks whether a given problem can be solved with this optimizer.
+
+        Checks whether the given problem is compatible, i.e., whether the problem can be converted
+        to a QUBO, and otherwise, returns a message explaining the incompatibility.
+
+        Args:
+            problem: The optimization problem to check compatibility.
+
+        Returns:
+            A message describing the incompatibility.
+        """
+        return QuadraticProgramToQubo.get_compatibility_msg(problem)
+
+    def solve_relaxed(
+        self,
+        encoding: QuantumRandomAccessEncoding,
+    ) -> Tuple[MinimumEigensolverResult, RoundingContext]:
+        """Solve the relaxed Hamiltonian given by the encoding.
+
+        Args:
+            encoding: The ``QuantumRandomAccessEncoding``, which must have already been ``encode()``ed
+                with a ``QuadraticProgram``.
+
+        Returns:
+            The result of the minimum eigensolver, and the rounding context.
+
+        Raises:
+            ValueError: If the encoding has not been encoded with a ``QuadraticProgram``.
+        """
+        if not encoding.frozen:
+            raise ValueError(
+                "The encoding must call ``encode()`` with a ``QuadraticProgram`` before being passed"
+                "to the QuantumRandomAccessOptimizer."
+            )
+
+        # Get the list of operators that correspond to each decision variable.
+        variable_ops = [encoding._term2op(i) for i in range(encoding.num_vars)]
+
+        # Solve the relaxed problem
+        start_time_relaxed = time.time()
+        relaxed_result = self.min_eigen_solver.compute_minimum_eigenvalue(
+            encoding.qubit_op, aux_operators=variable_ops
+        )
+        relaxed_result.time_taken = time.time() - start_time_relaxed
+
+        # Get auxiliary expectation values for rounding.
+        if relaxed_result.aux_operators_evaluated is not None:
+            expectation_values = [v[0] for v in relaxed_result.aux_operators_evaluated]
+        else:
+            expectation_values = None
+
+        # Get the circuit corresponding to the relaxed solution.
+        if hasattr(self.min_eigen_solver, "ansatz"):
+            circuit = self.min_eigen_solver.ansatz.bind_parameters(relaxed_result.optimal_point)
```

Comment:
Now in Terra we have a VariationalAlgorithm and VariationalResult. Arguably, it would be better to check the relaxed_result type and if its a VariationalResult then use the optimal_circuit from that result (I have no idea why it got called optimal circuit as its not the bound circuit just the original - it still needs the parameters bound as you have done). It would avoid knowing the circuit is passed by some ansatz field though.